react/CategoryItem.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { graphql } from 'react-apollo'
+import classNames from 'classnames'
+
+import categoryQuery from './queries/categoryQuery.gql'
+
+/**
+ * Component that represents a single category displayed in the menu, also displays
+ * the subcategories, if the provided category has them
+ */
+class CategoryItem extends Component {
+  render() {
+    const { data: { category } } = this.props
```

Comment:
In some cases the graphql query might return a `loading` status, before it has any category data available.
Add a fallback condition for this case, otherwise when the render method tries to access one of the category`s attribute it will thrown an exception.